Голосование за тип раунда

Kocma-san opened the pull request:
## О Pull Request

Добавляет голосование за тип раунда.
3 типа раунда:
- Зеленый - почти экста, минимум трита, которого хватит разве что на 1 антагониста.
- Синий - классический раунд, которые были всегда.
- Красный - жесткий динамик, всегда 85-100 трита + случайный раундстарт мажор (нюка/культ/рева).

И один секретный режим

Влияние на раунд будет дорабатываться. Текущая версия тестовая и является *не самым хорошим* куском кода. 

При добавлении конфига на сервер - нужно раскоментировать флаг `SHIFT_INTENSITY`. Не хочу чтобы эта система была по умолчанию на локалках людей
Заказ от Птички.

## Как это может улучшить/повлиять на игровой процесс/ролевую игру

Больше контроля у игроков за интенсивностью раунда. Игроки могут выбрать то, что им хочется от следующего раунда.

## Доказательства тестирования

<details>
<summary>Скриншоты/Видео</summary>

![image](https://github.com/user-attachments/assets/eb811eb8-7ed7-4098-909e-33b1a1fd6ea2)
![image](https://github.com/user-attachments/assets/5f24f517-d4a2-4cf7-a6bf-74b55d0b7b0e)
![image](https://github.com/user-attachments/assets/270ab310-a9f0-4fa5-a26c-fe0e31714ba7)

</details>

## Changelog

:cl:
add: Added shift intensity system.
add: Добавлен 4 уровень сложности смены.
/:cl:

Kocma-san committed:
v0.5

Какая-то штука. По идее что-то делает. Наверное

Iajret commented on tff_modular/modules/shift_intensity/shift_intensity_vote.dm:
```diff
@@ -0,0 +1,78 @@
+		ROUND_MID_SHIFT_STRING,
+		ROUND_HARD_SHIFT_STRING,
+	)
+	var/list/roundstart_rulesets = list(
```
```suggestion
	var/static/list/roundstart_rulesets = list(
```

Iajret reviewed (changes requested):
Ну чёт такое примерно вот так. Идея раундстарт мажоров мне абсолютно не нравится, потому что проблему с малым числом реди никто решать не хочет. Oh well.

Iajret committed:
Apply suggestions from code review

Kocma-san committed:
Update tff_modular/modules/shift_intensity/shift_intensity_vote.dm

Co-authored-by: Iajret <8430839+Iajret@users.noreply.github.com>

Kocma-san commented on tff_modular/modules/shift_intensity/shift_intensity_config.dm:
```diff
@@ -0,0 +1,14 @@
+// Определяет, работает ли SSshiftcolor или нет
+/datum/config_entry/flag/shift_intensity
+
+// Определяет, может ли голосование за shiftcolor быть начато кем-либо или нет
+/datum/config_entry/flag/allow_shift_intensity_vote
```
Первый флаг - отвечает за подсистему, которая сама запускает воут
Второй флаг - за то, могут ли игроки запустить голосование сами. Оно есть у всех голосовалок
По умолчанию оба выключены, но на фф первый нужно раскомментировать

Kocma-san commented on tff_modular/modules/shift_intensity/shift_intensity_vote.dm:
```diff
@@ -0,0 +1,75 @@
+
+/datum/vote/shift_intensity/finalize_vote(winning_option)
+	if(SSticker.current_state != GAME_STATE_PREGAME)
+		message_admins("Shift type vote ended after the round started. No changes to the round type. Current type: [winning_option]")
```
```suggestion
		message_admins("Shift type vote ended after the round started. No changes to the round type. Current type: [GLOB.shift_intensity_level]")
```

Kocma-san commented on tff_modular/modules/shift_intensity/shift_intensity_vote.dm:
```diff
@@ -0,0 +1,75 @@
+/datum/vote/shift_intensity/finalize_vote(winning_option)
+	if(SSticker.current_state != GAME_STATE_PREGAME)
+		message_admins("Shift type vote ended after the round started. No changes to the round type. Current type: [winning_option]")
+		log_admin("Shift type vote ended after the round started. No changes to the round type. Current type: [winning_option]")
```
```suggestion
		log_admin("Shift type vote ended after the round started. No changes to the round type. Current type: [GLOB.shift_intensity_level]")
```

Kocma-san committed:
Update tff_modular/modules/shift_intensity/shift_intensity_vote.dm

Kocma-san committed:
Update tff_modular/modules/shift_intensity/shift_intensity_vote.dm

Kocma-san commented on tff_modular/modules/shift_intensity/shift_intensity_vote.dm:
```diff
@@ -0,0 +1,75 @@
+
+/datum/vote/shift_intensity/finalize_vote(winning_option)
+	if(SSticker.current_state != GAME_STATE_PREGAME)
+		message_admins("Shift type vote ended after the round started. No changes to the round type. Current type: [GLOB.shift_intensity_level]")
+		log_admin("Shift type vote ended after the round started. No changes to the round type. Current type: [GLOB.shift_intensity_level]")
```
```suggestion
		message_admins("Shift type vote ended after the round started. No changes to the round type.")
		log_admin("Shift type vote ended after the round started. No changes to the round type.")
```

Kocma-san committed:
Update tff_modular/modules/shift_intensity/shift_intensity_vote.dm

Kocma-san committed:
verb for admins

Kocma-san committed:
weh (Теперь все привязано к конфигам)

Но конфиги недописаны пока
Так же в жесткий динамик теперь изменяются пороги лоу/мид/хайпопа у системы ICES

Kocma-san committed:
configs (probably bad)

Что-то натыкал, но понятия не имею что

Kocma-san committed:
Update tff_modular/modules/shift_intensity/shift_intensity_config.dm

Co-authored-by: Iajret <8430839+Iajret@users.noreply.github.com>

Kocma-san committed:
Update shift_intensity.dm

Kocma-san committed:
Merge branch 'vote-things' of https://github.com/Kocma-san/FluffySTG into vote-things

Kocma-san committed:
Update dynamic_hard.json

Kocma-san committed:
Merge remote-tracking branch 'upstream/master' into vote-things

Kocma-san committed:
v2

Kocma-san committed:
Merge branch 'master' into vote-things

Kocma-san committed:
fixes

Kocma-san committed:
update

Kocma-san committed:
Update dynamic.dm

Kocma-san committed:
v2.1

Kocma-san committed:
Update statpanel.dm

Kocma-san committed:
Update code/modules/mob/dead/new_player/login.dm

Kocma-san committed:
Update code/modules/mob/dead/new_player/login.dm

Kocma-san committed:
small addition

Kocma-san committed:
Update shift_intensity_vote.dm

Kocma-san committed:
Update shift_intensity.dm

Kocma-san committed:
Update shift_intensity.dm

Kocma-san committed:
Merge remote-tracking branch 'upstream/master' into vote-things

Kocma-san committed:
Merge remote-tracking branch 'upstream/master' into vote-things

Kocma-san committed:
Merge remote-tracking branch 'upstream/master' into vote-things

Kocma-san committed:
Update tgstation.dme

Kocma-san committed:
changes

Kocma-san committed:
Create readme.md

Kocma-san committed:
Update shift_intensity.dm
